examples/password-manager/src/4_actions.ts

```diff
@@ -54,3 +55,24 @@ export const shareFolder = (
   }
   return undefined;
 };
+
+export async function addSharedFolder(
+  sharedFolderId: ID<Folder>,
+  me: PasswordManagerAccount) {
+  const [sharedFolder, account] = await Promise.all([
+    await waitForCoValue(Folder, sharedFolderId, me, Boolean, {}),
+    await waitForCoValue(PasswordManagerAccount, me.id, me, Boolean, {
+      root: {
+        folders: [],
+      },
+    }),
+  ]);
```

Comment:
Used this helper to wait for the data to loaded, it works even if the user that triggered the sharing haven't uploaded the folder yet.